### PR TITLE
Fix drawing in df104 tutorial

### PR DESCRIPTION
**Pull request by linev**
1. Disable automatic drawing when fitting
2. Draw data histogram only when all settings are done
3. Use TH1I instead of TF1 for axis drawing in lower pad

Created canvas can be stored and read back properly from ROOT file

### CODE REVIEW

File: tutorials/dataframe/df104_HiggsToTwoPhotons.py [hunk 1]
```diff
@@ -175,7 +177,7 @@
 ratiobkg.GetYaxis().SetNdivisions(503, False)
 ratiobkg.GetYaxis().ChangeLabel(-1, -1, 0)
 ratiobkg.GetXaxis().SetTitle("m_{#gamma#gamma} [GeV]")
-ratiobkg.Draw()
+ratiobkg.Draw("AXIS")
```
Comment (couet) [inline tutorials/dataframe/df104_HiggsToTwoPhotons.py:180]: Usually the AXIS option is used internally only ...

Comment (couet) [reply tutorials/dataframe/df104_HiggsToTwoPhotons.py:180]: But fine ..

**Review by couet — APPROVED** [review]

File: tutorials/dataframe/df104_HiggsToTwoPhotons.py (hunk 1 above)
Comment (eguiraud) [reply tutorials/dataframe/df104_HiggsToTwoPhotons.py:180]: @linev following Olivier's comment, why do we need "AXIS" here?

Comment (linev) [reply tutorials/dataframe/df104_HiggsToTwoPhotons.py:180]: "AXIS" draw only histogram axes without content. Also no stats box will be created.
And it is [fully legitime](https://root.cern/doc/master/classTHistPainter.html#HP01a) draw option

Comment (eguiraud) [reply tutorials/dataframe/df104_HiggsToTwoPhotons.py:180]: Ok then I'm confused by @couet 's statement :smile:

**Review by eguiraud — APPROVED** [review]
not a graphics expert, but I trust our experts!

File: tutorials/dataframe/df104_HiggsToTwoPhotons.py (hunk 1 above)
Comment (linev) [reply tutorials/dataframe/df104_HiggsToTwoPhotons.py:180]: `TF1` internally uses exactly this method to draw axes. Problem of solution with `TF1` - not all properties will be correctly preserved when `TCanvas` stored in the ROOT file. Therefore I would always recommend to use `TH1` class for the case when just axis should be drawn

**Pull request merged**